geode-wan/src/main/java/org/apache/geode/internal/cache/wan/GatewayReceiverImpl.java

```diff
@@ -162,16 +162,9 @@ public void start() throws IOException {
         receiver.start();
         started = true;
       } catch (BindException be) {
-        if (be.getCause() != null
-            && be.getCause().getMessage().contains("assign requested address")) {
-          throw new GatewayReceiverException(
-              LocalizedStrings.SocketCreator_FAILED_TO_CREATE_SERVER_SOCKET_ON_0_1
-                  .toLocalizedString(new Object[] {bindAdd, this.port}));
-        }
-        // ignore as this port might have been used by other threads.
-        logger.warn(LocalizedMessage.create(LocalizedStrings.GatewayReceiver_Address_Already_In_Use,
-            this.port));
-        this.port = getPortToStart();
+        throw new GatewayReceiverException(
```

Comment:
I'm not certain about the behavior change, but this code hides the originating exception.  We should use the pattern:

```
catch (Exception e) {
  throw new Exception("Some message", e);
}
```

Reply:
I think it is okay to hide the stack trace since we understand why a BindException can take place. They gave us an address or port that was already in use. So I like that in the GatewayReceiverException we include the "bindAdd" and "this.port". I think we would only need the stack trace from the BindException (by including ", e") if we didn't expect the exception in this context. But I think we should include the message from the BindException as it may give the user additional clues as to why it could not bind the socket.

Reply:
There are two possible reasons why the socket couldn't bind: permissions and something got there first. The fact the exception happened at all seems to me to be sufficient for the user to troubleshoot. Adding the error message (which will be either "permission denied" or "can't bind" or some similar language) feels like it would make the code a lot more cumbersome and not provide much value.

Reply:
But with your last change, adding ", be", you are adding a cause to the exception which can make it even more cumbersome. I guess it all depends on how this exception ends up getting handled by our code. Usually gateways are started when a server is started so at some point that code logs this exception. Too much info is better than too little that is why I'm okay with your latest change